src/gsy_e/gsy_e_core/user_profile_handler.py

```diff
@@ -341,3 +353,21 @@ def should_create_profile(self, profile):
     def get_profile_type(self, profile_uuid: str) -> InputProfileTypes:
         """Read the profile type from a profile with the specified UUID."""
         return self.db.get_profile_type_from_db_buffer(profile_uuid)
+
+    def _get_profile_uuids_from_setup(self, area: "Area", profile_uuids: List) -> None:
```

Comment:
https://github.com/gridsingularity/gsy-e/pull/1550#discussion_r994655107 and this is connected. I will come up with a recursive function that has a return value and not use assing-by-refference.

Reply:
I refactored the function and added two unit tests for catching this bug here: [ac0127e](https://github.com/gridsingularity/gsy-e/pull/1550/commits/ac0127ecfebe170635ea051abc8bd26452037b7e)

I spend a lot of time adding the unit tests because I tried to mock a DB connection to pony. I was not successful and therefore went with mocking parts of `ProfileDBConnectionHandler`.
The tests also include checking the case when there is a profile missing in the DB, which is the same check as in the integration test I added. Shall I keep both in order to also test the whole workflow with a DB connection or shall I only keep the unit test?
@spyrostz @mrsaemir

Reply:
@hannesdiedrich IMO I would very much like to have both, one unit test for quick verification, and the integration test that verifies that it is working as expected with a real DB